pkg/scheduler/framework/plugins/noderesources/resource_allocation.go

```diff
@@ -109,28 +111,7 @@ func (r *resourceAllocationScorer) calculateResourceAllocatableRequest(nodeInfo
 // calculatePodResourceRequest returns the total non-zero requests. If Overhead is defined for the pod
 // the Overhead is added to the result.
 // podResourceRequest = max(sum(podSpec.Containers), podSpec.InitContainers) + overHead
-func (r *resourceAllocationScorer) calculatePodResourceRequest(pod *v1.Pod, resource v1.ResourceName) int64 {
-	var podRequest int64
-	for i := range pod.Spec.Containers {
-		container := &pod.Spec.Containers[i]
-		value := schedutil.GetRequestForResource(resource, &container.Resources.Requests, !r.useRequested)
-		podRequest += value
-	}
-
-	for i := range pod.Spec.InitContainers {
-		initContainer := &pod.Spec.InitContainers[i]
-		value := schedutil.GetRequestForResource(resource, &initContainer.Resources.Requests, !r.useRequested)
-		if podRequest < value {
-			podRequest = value
-		}
-	}
-
-	// If Overhead is being utilized, add to the total requests for the pod
-	if pod.Spec.Overhead != nil {
-		if quantity, found := pod.Spec.Overhead[resource]; found {
-			podRequest += quantity.Value()
-		}
-	}
-
-	return podRequest
+func (r *resourceAllocationScorer) calculatePodResourceRequest(pod *v1.Pod, name v1.ResourceName) int64 {
+	requests, _ := resource.PodRequestsAndLimits(pod)
+	return schedutil.GetRequestForResource(name, &requests, !r.useRequested)
```

Comment:
does it change the result when multiple pods don't have requests specified and use defaults instead. In previous implementation it will be sum of defaults, not a single default value

Reply:
if this will not be caught by tests, it would be nice to add a test like this

Reply:
Yup, it's actually caught by an existing test.  I thought I had reverted that change in this PR, but apparently missed it.  I put up https://github.com/kubernetes/kubernetes/pull/115368 to correct the comment that the defaults are added per container vs per pod.

Reply:
would it make sense to generalize the underlying function to allow to call it from here as well?

Reply:
Potentially, I was going to de-dupe the straight forward ones first and then re-visit.   I am interested in your opinion on:
- Is the interface for PodRequestsAndLimits that's already exported unchangable?  I'm not sure of what the Kubernetes compatibility promise is for exported functions.
- How do you feel about the waste here?  We're computing limits and then throwing them away. I could see a case being made for a more generic function that can handle all of these different use cases efficiently.